src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -469,6 +562,307 @@ class IpexCommunicationService extends AgentService {
       await this.notificationStorage.update(notificationRecord);
     }
   }
+
+  @OnlineOnly
+  async joinMultisigOffer(said: string): Promise<void> {
```

Comment:
I think `joinMultisigOffer` and `joinMultisigGrant` can be private right? If so we can remove the `@OnlineOnly` decorator